### PR TITLE
feat: Enhanced Bulk Minting: Add Support for Onchain Attributes

### LINKED ISSUE #11149: Enhanced Bulk Minting: Add Support for Onchain Attributes
Extend the current bulk minting functionality to allow users to add onchain attributes to NFTs during the minting process. These attributes should be stored on-chain and be available for querying and displaying.

### PR DESCRIPTION
**Pull request by Jarsen136**
**Thank you for your contribution** to the [Koda - Generative Art Marketplace](https://kodadot.xyz).

👇 __ Let's make a quick check before the contribution.

## PR Type

- [x] Bugfix
- [x] Feature
- [x] Supported description file type: json and txt. 
- CSV continued to https://github.com/kodadot/nft-gallery/issues/11282
- [x] Refactor

## Needs Design check

- @exezbcz please review

## Needs QA check

- @kodadot/qa-guild please review

## Context

- [x] Closes #11149

## Screenshot 📸

- [x] My fix has changed **something** on UI; a screenshot is best to understand changes for others.

image
image
image
image

minted nft example: https://canary.kodadot.xyz/ahk/gallery/462-9 \ https://canary.kodadot.xyz/ahk/gallery/462-10

### CODE REVIEW

**Review by hassnian — CHANGES_REQUESTED** [review]
1. modal seems to be broken

CleanShot 2024-12-22 at 07 31 57@2x

2. attributes component not working if the item is not populated

https://github.com/user-attachments/assets/c9470841-cc04-4934-86c9-ff81a05c1c97

**Comment by Jarsen136** [comment]
> 1. modal seems to be broken
> 
> 
> 2. attributes component not working if the item is not populated
> 

✅ All fixed

File: components/massmint/EditPanel.vue [hunk 1]
```diff
@@ -100,6 +107,7 @@ import {
   NeoSidebar,
 } from '@kodadot1/brick'
 import type { NFT } from './types'
+import CustomAttributeInput from '@/components/rmrk/Create/CustomAttributeInput.vue'
```
Comment (hassnian) [inline components/massmint/EditPanel.vue:110]: we should slowly get rid off the `rmrk` folder

File: components/massmint/types.ts [hunk 2]
```diff
@@ -16,15 +16,15 @@ export type NFT = {
   description?: string
   price?: number
   status?: Status
-  attributes?: OpenSeaAttribute[]
+  tags?: OpenSeaAttribute[]
```
Comment (hassnian) [inline components/massmint/types.ts:19]: why the change to `tags` I'd keep it as `attributes`

File: composables/massmint/parsers/parseTxt.ts [hunk 3]
```diff
@@ -91,6 +94,7 @@ const updateEntries = (entries, block) => {
       file: entry.file,
       name: entry.name,
       description: entry.description,
+      tags: entry.attributes,
```
Comment (hassnian) [inline composables/massmint/parsers/parseTxt.ts:97]: `tags` and `attribute` let's keep one unless there is reason

**Review by hassnian — COMMENTED** [review]
otherwise LGTM

File: components/massmint/EditPanel.vue (hunk 1 above)
Comment (Jarsen136) [reply components/massmint/EditPanel.vue]: done https://github.com/kodadot/nft-gallery/pull/11267/commits/71bbd1b82dcb50c4d443b7e537168faad8a8015e

File: composables/massmint/parsers/parseTxt.ts (hunk 3 above)
Comment (Jarsen136) [reply composables/massmint/parsers/parseTxt.ts:97]: I guess the `attribute` is more user-friendly and easy to understand, so I put `attributes` file in the template file. However, when we call the polkdot API, we must use `tags` in the nft instead.

File: components/massmint/types.ts (hunk 2 above)
Comment (Jarsen136) [reply components/massmint/types.ts:19]: same reason as below comment

**Review by preschian — APPROVED** [review]

File: composables/massmint/parsers/parseTxt.ts (hunk 3 above)
Comment (hassnian) [reply composables/massmint/parsers/parseTxt.ts:97]: Fair enough, but I would still keep one naming across the app **just for the sake of consistency** in this case attributes, since its already used in other places

See https://github.com/kodadot/nft-gallery/blob/main/components/common/EditNftModal.vue

And tags which is specific of SUB can be used inside transaction token mint logic

Let me know if it makes sense

Comment (Jarsen136) [reply composables/massmint/parsers/parseTxt.ts]: Alright, I have updated it.

**Review by hassnian — APPROVED** [review]
